src/nebula/NebulaContext.cs

```diff
@@ -25,12 +26,28 @@ public NebulaContext()
 
         internal IComponentContext ComponentContext { get; set; }
 
+        public string MongoConnectionString { get; set; }
+        public string RedisConnectionString { get; set; }
+        
         public void RegisterJobProcessor(Type processor, Type stepType)
         {
             var contract = typeof(IJobProcessor<>).MakeGenericType(stepType);
             ComponentContext.Register(contract, processor);
         }
 
+        public void RegisterJobProcessor(object processor, Type stepType)
+        {
+            var contract = typeof(IJobProcessor<>).MakeGenericType(stepType);
+
+            var isTypeCorrect = processor.GetType().GetInterfaces().Any(x =>
+                x.IsGenericType && x.GetGenericTypeDefinition() == typeof(IJobProcessor<>));
+
+            if (!isTypeCorrect)
+                throw new ArgumentException("processor should implement IJobProcessor<>");
```

Comment:
Start the error sentence with capital letter - "Processor" instead of "processor"